javaslang/src/main/java/javaslang/collection/CharSeq.java

```diff
@@ -1439,14 +1059,25 @@ public boolean contentEquals(CharSequence cs) {
      * produces the same result
      * </ul>
      *
-     * @param anotherString The {@code CharSeq} to compare this {@code CharSeq} against
+     * @param that The {@code CharSeq} to compare this {@code CharSeq} against
      * @return {@code true} if the argument is not {@code null} and it
      * represents an equivalent {@code CharSeq} ignoring case; {@code
      * false} otherwise
      * @see #equals(Object)
      */
-    public boolean equalsIgnoreCase(CharSeq anotherString) {
-        return back.equalsIgnoreCase(anotherString.back);
+    public boolean equalsIgnoreCase(CharSeq that) {
```

Comment:
Could be `delegate.map(Character::toLowerCase).equals(that.delegate.map(Character::toLowerCase)) also, but this should be a lot more optimal (it can short circuit in case of non-equality, and only converts to lowercase if not equal)